### PR TITLE
created ExploreFragment and it's logic part.

### LINKED ISSUE #99: Creating and implementing explore fragment.
I want to contribute to this functionality. I'll be storing the data in Firebase firestore and create a fragment with a search bar that will provide the resources stored on firebase for that particular tech.
For the representation, I'll be using a recycler view to show the data.
For future updates, I can add images too in the recycler view.

### PR DESCRIPTION
**Pull request by ayushman999**
## Explanation

Fixes #99 
This PR completes the explore fragment logic part that is #99  and fetches data from the firestore and shows it in a different activity.
## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR does not break any part of the existing code.
- [x] The PR creator have carefully read & adher to the contributing guidelines of the project.
- [x] The PR is made to the branch that's **called** "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "main" and not on any other branch.

### CODE REVIEW

File: app/src/main/java/com/cswala/cswala/Fragments/ExploreFragment.java [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull @NotNull Exception e) {
+                Toast.makeText(getContext(), "Something went wrong...", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+    @Override
+    public void onItemClicked(String item) {
```
Comment (riturajjain2000) [inline app/src/main/java/com/cswala/cswala/Fragments/ExploreFragment.java:88]: Add this intent to intent helper class

File: app/src/main/java/com/cswala/cswala/Fragments/ExploreFragment.java [hunk 2]
```diff
@@ -1,24 +1,93 @@
...
+        Intent transfer=new Intent(getContext(), TechDataActivity.class);
```
Comment (riturajjain2000) [inline app/src/main/java/com/cswala/cswala/Fragments/ExploreFragment.java:89]: Add this intent to intent helper class

**Review by riturajjain2000 — CHANGES_REQUESTED** [review]
- Remove the 'EXPLORE ' text from the top of the fragment
- The search bar is not visible , add a  background 
- add a small secondary text in the list items in the upper right corner , which will show the 'tag' string from the DB

Rest LGTM!

File: app/src/main/java/com/cswala/cswala/Fragments/ExploreFragment.java (hunk 2 above)
Comment (ayushman999) [reply app/src/main/java/com/cswala/cswala/Fragments/ExploreFragment.java:89]: Sure

**Comment by ayushman999** [comment]
Did all the required changes, please go through it.

**Review by riturajjain2000 — CHANGES_REQUESTED** [review]
- make the search icon, placeholder text and, close icon color dark, as white on yellow is not visible, also see if you can make the placeholder 'search...' to match the same font as the other texts in the app i.e to `press start 2p`
- You added the `tag` in the tech data activity , you have to add that in the list items in the explore fragment 
Screenshot_1616391361

like this in the upper right corner 

- The app is crashing now , some problem in the Animation of login class because of missing handlers for thread , so fix that in this PR only and raise a issue for same so as you can get points

**Comment by riturajjain2000** [comment]
### LOGCAT -
`android.util.AndroidRuntimeException: Animators may only be run on Looper threads
            at android.animation.ValueAnimator.cancel(ValueAnimator.java:1104)
            at android.view.ViewPropertyAnimator.animatePropertyBy(ViewPropertyAnimator.java:956)
            at android.view.ViewPropertyAnimator.animateProperty(ViewPropertyAnimator.java:912)
            at android.view.ViewPropertyAnimator.alpha(ViewPropertyAnimator.java:729)
            at com.cswala.cswala.Activities.LoginActivity$6.run(LoginActivity.java:218)
            at java.util.TimerThread.mainLoop(Timer.java:562)
            at java.util.TimerThread.run(Timer.java:512)`

**Review by riturajjain2000 — APPROVED** [review]

**Pull request merged**